aif360/aif360-r/R/inprocessing_adversarial_debiasing.R

```diff
@@ -1,40 +1,73 @@
+#' @param unprivileged_groups A list with two values: the column of the protected class and the value indicating representation for unprivileged group.
+#' @param privileged_groups A list with two values: the column of the protected class and the value indicating representation for privileged group.
+#' @param scope_name Scope name for the tensorflow variables.
 #' @param sess tensorflow session
-#' @param seed seed to make `predict` repeatable.
-#' @param adversary_loss_weight hyperparameter that chooses the strength of the adversarial loss.
-#' @param num_epochs number of training epochs.
-#' @param batch_size batch size.
-#' @param classifier_num_hidden_units number of hidden units in the classifier model.
-#' @param debias learn a classifier with or without debiasing.
+#' @param seed Seed to make \code{predict} repeatable. If not, \code{NULL}, must be an integer.
+#' @param adversary_loss_weight Hyperparameter that chooses the strength of the adversarial loss.
+#' @param num_epochs Number of training epochs. Must be an integer.
+#' @param batch_size Batch size. Must be an integer.
+#' @param classifier_num_hidden_units Number of hidden units in the classifier model. Must be an integer.
+#' @param debias Learn a classifier with or without debiasing.
 #' @examples
 #' \dontrun{
 #' load_aif360_lib()
 #' ad <- adult_dataset()
 #' p <- list("race", 1)
 #' u <- list("race", 0)
 #'
-#' sess = tf$compat$v1$Session()
+#' sess <- tf$compat$v1$Session()
 #'
-#' plain_model = adversarial_debiasing(privileged_groups = p,
-#'                                     unprivileged_groups = u,
-#'                                     scope_name='debiased_classifier',
-#'                                     debias=TRUE,
-#'                                     sess=sess)
+#' plain_model <- adversarial_debiasing(privileged_groups = p,
+#'                                      unprivileged_groups = u,
+#'                                      scope_name = "debiased_classifier",
+#'                                      debias = TRUE,
+#'                                      sess = sess)
 #'
 #' plain_model$fit(ad)
 #' ad_nodebiasing <- plain_model$predict(ad)
 #' }
 #' @export
 #'
-adversarial_debiasing <- function(unprivileged_groups, privileged_groups, scope_name='current', sess=tf$compat$v1$Session(),
-                                  seed=NULL, adversary_loss_weight=0.1, num_epochs=50, batch_size=128,
-                                  classifier_num_hidden_units=200, debias=TRUE) {
+adversarial_debiasing <- function(unprivileged_groups, 
+                                  privileged_groups, 
+                                  scope_name = "current", 
+                                  sess = tf$compat$v1$Session(),
+                                  seed = NULL, 
+                                  adversary_loss_weight = 0.1, 
+                                  num_epochs = 50L, 
```

Comment:
I think it should be 50 and 128. Can you refer here?
https://github.com/Trusted-AI/AIF360/blob/bdc53541e9b4cfcfd6d1bf7923e2f147c66a1085/aif360/algorithms/inprocessing/adversarial_debiasing.py#L34

Reply:
Yes, in Python `50` will get treated as an integer, but in R, `50` gets treated as a double. This raises an error when running the `adversarial_debiasing()`. Setting them as e.g., `50L` ensures the arguments are treated as integers and avoids raising a Python error.

Reply:
Tested and works fine